### PR TITLE
feat: integration configuration

### PR DESCRIPTION
**Pull request by naftis**
This pull request is to implement the tickets in epic https://github.com/opencrvs/opencrvs-core/issues/2277
Farajaland PR: https://github.com/opencrvs/opencrvs-farajaland/pull/390

Before merging:
- [x] Make sure we have captured all the translations in English & French. Run extract translations

### CODE REVIEW

**Review by Zangetsu101 — APPROVED** [review]

**Comment by euanmillar** [comment]
@naftis we need to make a new migration for the user-mgnt system model.  The name, createdBy and scope=>type props need migrated from any entries in the old collection to the new collection.  Please can you do this in a separate pull request?

File: packages/client/src/views/SysAdmin/Config/Systems/Systems.tsx [hunk 1]
```diff
@@ -663,7 +667,7 @@ export function SystemList() {
                               label: intl.formatMessage(
                                 integrationMessages.documentDetails
                               ),
-                              value: 'documents'
+                              value: 'supporting-documents'
```
Comment (Zangetsu101) [inline packages/client/src/views/SysAdmin/Config/Systems/Systems.tsx:670]: @euanmillar These 3 values need to be updated in `WebhookModal.tsx` as well, similar to this commit:https://github.com/opencrvs/opencrvs-core/pull/4138/commits/8084dbe0a613507a5964702ce7a00687680b3e18

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/user-mgnt/src/model/system.ts [hunk 2]
```diff
@@ -26,15 +25,40 @@ export interface ISystem {
...
+  permissions: { type: [String] }
+})
+
 export interface ISystemModel extends ISystem, Document {}
 
 const systemSchema = new Schema({
-  name: { type: [UserNameSchema], required: true },
-  createdBy: { type: [UserNameSchema], required: true },
+  name: { type: String, required: true },
+  createdBy: { type: String, required: true },
```
Comment (naftis) [inline packages/user-mgnt/src/model/system.ts:61]: @euanmillar @Zangetsu101 

1) this `createdBy` could be a reference to User-model id: https://mongoosejs.com/docs/populate.html
2) Are Systems always before 1.2.0, and for now for upcoming versions, always made by the single natlsysadmin? I'm thinking about migrations. Can we just set the `createdBy` to be the single natlsysadmin for now?

If multiple users have been able to create systems earlier, I think there are destructive ways only to migrate this.

File: packages/user-mgnt/src/model/system.ts [hunk 3]
```diff
@@ -48,9 +72,14 @@ const systemSchema = new Schema({
     default: statuses.PENDING
   },
   settings: {
-    dailyQuota: { type: String, default: 0 }
+    webhook: { type: [WebhookSchema], required: false },
+    dailyQuota: { type: Number, default: 0 }
```
Comment (naftis) [inline packages/user-mgnt/src/model/system.ts:76]: This only needs a down migration where webhooks are removed.

File: packages/user-mgnt/src/model/system.ts [hunk 4]
```diff
@@ -48,9 +72,14 @@ const systemSchema = new Schema({
...
   },
-  creationDate: { type: Number, default: Date.now }
+  creationDate: { type: Number, default: Date.now },
+  type: {
```
Comment (naftis) [inline packages/user-mgnt/src/model/system.ts:79]: Type should be derived from `scope`

File: packages/user-mgnt/src/model/system.ts (hunk 2 above)
Comment (naftis) [reply packages/user-mgnt/src/model/system.ts:61]: The OpenCRVS documentation requires you to SSH in so you must be a system administrator. I reflected this here https://github.com/opencrvs/opencrvs-core/pull/4444/commits/c3b4c0f2f5089e32d14777978735b86fbcb41a77

Comment (euanmillar) [reply packages/user-mgnt/src/model/system.ts:61]: Before 1.2.0 it is always only the national system admin. I took a look at the PR you did for it and it looked good to me

**Pull request merged**